src/lib/components/Search/index.js

```diff
@@ -0,0 +1,347 @@
+                id="web-search-popout__list"
+                class="web-search-popout__list"
+                role="listbox"
+              >
+                ${this.hits.map(
+                  (hit, idx) => html`
+                    <li class="web-search-popout__item">
+                      <a
+                        id="web-search-popout__link--${idx}"
+                        class="web-search-popout__link ${idx === this.cursor
+                          ? "web-search-popout__link--active"
+                          : ""}"
+                        aria-selected="${idx === this.cursor}"
+                        href="${hit.url}"
+                        >${hit.title}</a
+                      >
+                    </li>
+                  `,
+                )}
+              </ul>
+            </div>
+          `
+        : ""}
+    `;
+  }
+
+  /**
+   * Keep track of cursor changes and reflect them to aria-activedescendant.
+   * This ensures screen readers properly announce the current search result.
+   * We do this because focus never leaves the search input box, so when the
+   * user is arrowing through results, we have to tell the screen reader about
+   * it.
+   * @param {Map} changedProperties A Map of LitElement properties that changed.
+   */
+  updated(changedProperties) {
+    if (!changedProperties.has("cursor")) {
+      return;
+    }
+
+    const input = this.querySelector(".web-search__input");
+
+    if (this.cursor === -1) {
+      input.removeAttribute("aria-activedescendant");
+      return;
+    }
+
+    input.setAttribute(
+      "aria-activedescendant",
+      `web-search-popout__link--${this.cursor}`,
+    );
+  }
+
+  /**
+   * Grab the animation custom property to infer how long our javascript should
+   * wait before doing state changes.
+   */
+  onResize() {
+    const styles = getComputedStyle(this);
+    const value = String(
+      styles.getPropertyValue("--web-search-animation-time"),
+    ).trim();
+    // value will either be "200ms" or "0".
+    this.animationTime = Number(value.split("ms").shift());
+  }
+
+  onKeyUp(e) {
+    // Check if the user is navigating within the search popout.
+    switch (e.key) {
+      case "Home":
+        this.firstHit();
+        return;
+
+      case "End":
+        this.lastHit();
+        return;
+
+      case "ArrowUp":
+        this.prevHit();
+        return;
+
+      case "ArrowDown":
+        this.nextHit();
+        return;
+
+      case "Enter":
+        this.navigateToHit(this.hits[this.cursor]);
+        return;
+
+      case "Escape":
+        document.activeElement.blur();
+        return;
+    }
+
+    // If the user is not navigating within the search popout, then assume
+    // anything they type is part of their query and ping algolia.
+    const query = e.target.value;
+    if (query === "") {
+      this.hits = [];
+      return;
+    }
+    (async () => {
+      try {
+        const {hits} = await index.search({query, hitsPerPage: 10});
```

Comment:
I added a simple debounce util to avoid spamming algolia (and to use on the resize handler).
I also added a local copy of query to check against before assigning values.